### ISSUE TITLE
Generate CHANGELOG and package-lock.json updates again during the release CI

### ISSUE THREAD

**Issue by adamstankiewicz** [openedx/paragon#1768]
The `openedx/cla` status check was recently made required to ensure all pushes to `master` have a signed contributor agreement.

However, Paragon uses `semantic-release` to release to NPM and generate a CHANGELOG.

The issue that was blocking NPM releases was our usage of `@semantic-release/git` which attempts to commit changes to CHANGELOG file and package.json to `master`, but fails due to `openedx/cla` status check.

As an [interim solution](https://github.com/openedx/paragon/pull/1760) to unblock releases, we removed the `@semantic-release/git` plugin. This made releases work again (since that is done by `@semantic-release/github` to create GH releases/tags).

The consequence of that interim solution is that we're no longer generating CHANGELOG changes and/or committing updates to the package.json file after a PR is merged to `master`.

**AC**
- [ ] Ensure CHANGELOG entries are generated again after a PR is successfully released to NPM.
- [ ] Ensure package.json file is updated with new SemVer version after a PR is successfully released to NPM.
- [ ] Ensure file changes (CHANGELOG, package.json) is committed to `master` after a PR is successfully released to NPM.

**Implementation notes**
* Likely add back `@semantic-release/changelog` again to generate the CHANGELOG entries based on the commit(s) and update the file.
* Look into alternative ways to commit updates to CHANGELOG and package.json to `master`.
  * Consider something along the lines of how the dependent-usage-analyzer opens/auto-merges a PR ([example](https://github.com/openedx/paragon/blob/master/.github/workflows/analyze-dependents.yml#L259-L278)).

**Comment by viktorrusakov** [comment]
Closing this a duplicate of https://github.com/openedx/paragon/issues/2575
